packages/contract/lib/handlers.js

```diff
@@ -41,13 +41,38 @@ const handlers = {
    * @param {Object}       context  execution state
    * @param {PromiEvent}   emitter  promiEvent returned by a web3 method call
    */
-  setup: function(emitter, context) {
+  setup: function (emitter, context) {
     emitter.on("error", handlers.error.bind(emitter, context));
     emitter.on("transactionHash", handlers.hash.bind(emitter, context));
+
     // web3 block polls if the confirmation listener is enabled so we want to
     // give users a way of opting out of this behavior - it causes problems in testing
     if (!context.contract.disableConfirmationListener) {
-      emitter.on("confirmation", handlers.confirmation.bind(emitter, context));
+      // maintain a count of confirmation listeners
```

Comment:
TODO: expand comment to explain the mechanics of how this impacts tests with eager mode.